### ISSUE TITLE
Build error: entrypoint.entrypoint() is removed because of a bug

### ISSUE THREAD

**Issue by jmglov** [pixie-lang/pixie#463]
Compiling fails with the latest pypy; see error below. I'll hack the build script to use a release version, as I sucessfully built two or three weeks ago, and report back.

```
: jmglov@laurana; make build_with_jit
mkdir externals; \
cd externals; \
curl https://bitbucket.org/pypy/pypy/get/default.tar.bz2 >  pypy.tar.bz2; \
mkdir pypy; \
cd pypy; \
tar -jxf ../pypy.tar.bz2 --strip-components=1
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 16.2M  100 16.2M    0     0  1102k      0  0:00:15  0:00:15 --:--:-- 1198k
echo https://github.com/pixie-lang/external-deps/releases/download/1.0/`uname -s`-`uname -m`.tar.bz2
https://github.com/pixie-lang/external-deps/releases/download/1.0/Linux-x86_64.tar.bz2
curl -L https://github.com/pixie-lang/external-deps/releases/download/1.0/`uname -s`-`uname -m`.tar.bz2 > /tmp/externals.tar.bz2
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   597    0   597    0     0    941      0 --:--:-- --:--:-- --:--:--   941
100  517k  100  517k    0     0   206k      0  0:00:02  0:00:02 --:--:--  441k
tar -jxf /tmp/externals.tar.bz2 --strip-components=2
[translation:info] 2.7.9 (default, Oct  7 2015, 15:29:54)
[translation:info] [GCC 4.8.5]
[platform:msg] Set platform with 'host' cc=None, using cc='gcc', version='Unknown'
[translation:info] Translating target as defined by target
[platform:execute] gcc -c -O3 -pthread -fomit-frame-pointer -Wall -Wno-unused /tmp/usession-default-0/gcctest.c -o /tmp/usession-default-0/gcctest.o
[platform:execute] gcc /tmp/usession-default-0/gcctest.o -pthread -Wl,--export-dynamic -lrt -o /tmp/usession-default-0/gcctest
[platform:execute] gcc -c -O3 -pthread -fomit-frame-pointer -Wall -Wno-unused /tmp/usession-default-0/gcctest.c -o /tmp/usession-default-0/gcctest.o
[platform:execute] gcc /tmp/usession-default-0/gcctest.o -pthread -Wl,--export-dynamic -lrt -o /tmp/usession-default-0/gcctest
[platform:execute] gcc -c -O3 -pthread -fomit-frame-pointer -Wall -Wno-unused /tmp/usession-default-0/platcheck_0.c -o /tmp/usession-default-0/platcheck_0.o
[platform:execute] gcc /tmp/usession-default-0/platcheck_0.o -pthread -Wl,--export-dynamic -lrt -o /tmp/usession-default-0/platcheck_0
[platform:execute] gcc -c -O3 -pthread -fomit-frame-pointer -Wall -Wno-unused /tmp/usession-default-0/platcheck_1.c -o /tmp/usession-default-0/platcheck_1.o
[platform:execute] gcc /tmp/usession-default-0/platcheck_1.o -pthread -Wl,--export-dynamic -lintl -lrt -o /tmp/usession-default-0/platcheck_1
[platform:execute] gcc -c -O3 -pthread -fomit-frame-pointer -Wall -Wno-unused /tmp/usession-default-0/platcheck_2.c -o /tmp/usession-default-0/platcheck_2.o
[platform:execute] gcc /tmp/usession-default-0/platcheck_2.o -pthread -Wl,--export-dynamic -lrt -o /tmp/usession-default-0/platcheck_2
[platform:execute] gcc -c -O3 -pthread -fomit-frame-pointer -Wall -Wno-unused /tmp/usession-default-0/platcheck_3.c -o /tmp/usession-default-0/platcheck_3.o
[platform:execute] gcc /tmp/usession-default-0/platcheck_3.o -pthread -Wl,--export-dynamic -lrt -o /tmp/usession-default-0/platcheck_3
[platform:execute] gcc -c -O3 -pthread -fomit-frame-pointer -Wall -Wno-unused /tmp/usession-default-0/platcheck_4.c -o /tmp/usession-default-0/platcheck_4.o
[platform:execute] gcc /tmp/usession-default-0/platcheck_4.o -pthread -Wl,--export-dynamic -lrt -o /tmp/usession-default-0/platcheck_4
[platform:execute] gcc -c -O3 -pthread -fomit-frame-pointer -Wall -Wno-unused /tmp/usession-default-0/platcheck_5.c -o /tmp/usession-default-0/platcheck_5.o
[platform:execute] gcc /tmp/usession-default-0/platcheck_5.o -pthread -Wl,--export-dynamic -lrt -o /tmp/usession-default-0/platcheck_5
[platform:execute] gcc -c -O3 -pthread -fomit-frame-pointer -Wall -Wno-unused /tmp/usession-default-0/platcheck_6.c -o /tmp/usession-default-0/platcheck_6.o
[platform:execute] gcc /tmp/usession-default-0/platcheck_6.o -pthread -Wl,--export-dynamic -lrt -o /tmp/usession-default-0/platcheck_6
[platform:execute] gcc -c -O3 -pthread -fomit-frame-pointer -Wall -Wno-unused /tmp/usession-default-0/platcheck_7.c -o /tmp/usession-default-0/platcheck_7.o
[platform:execute] gcc /tmp/usession-default-0/platcheck_7.o -pthread -Wl,--export-dynamic -lrt -o /tmp/usession-default-0/platcheck_7
[platform:execute] gcc -c -O3 -pthread -fomit-frame-pointer -Wall -Wno-unused /tmp/usession-default-0/platcheck_8.c -o /tmp/usession-default-0/platcheck_8.o
[platform:execute] gcc /tmp/usession-default-0/platcheck_8.o -pthread -Wl,--export-dynamic -lrt -o /tmp/usession-default-0/platcheck_8
[platform:execute] gcc -c -O3 -pthread -fomit-frame-pointer -Wall -Wno-unused /tmp/usession-default-0/platcheck_9.c -o /tmp/usession-default-0/platcheck_9.o
[platform:execute] gcc /tmp/usession-default-0/platcheck_9.o -pthread -Wl,--export-dynamic -lrt -o /tmp/usession-default-0/platcheck_9
[platform:execute] gcc -c -O3 -pthread -fomit-frame-pointer -Wall -Wno-unused /tmp/usession-default-0/platcheck_10.c -o /tmp/usession-default-0/platcheck_10.o
[platform:execute] gcc /tmp/usession-default-0/platcheck_10.o -pthread -Wl,--export-dynamic -ldl -lrt -o /tmp/usession-default-0/platcheck_10
[platform:execute] gcc -c -O3 -pthread -fomit-frame-pointer -Wall -Wno-unused /tmp/usession-default-0/platcheck_11.c -o /tmp/usession-default-0/platcheck_11.o
[platform:execute] gcc /tmp/usession-default-0/platcheck_11.o -pthread -Wl,--export-dynamic -lrt -o /tmp/usession-default-0/platcheck_11
[platform:execute] gcc -c -O3 -pthread -fomit-frame-pointer -Wall -Wno-unused -I/usr/lib64/libffi-3.0.13/include /tmp/usession-default-0/platcheck_12.c -o /tmp/usession-default-0/platcheck_12.o
[platform:execute] gcc /tmp/usession-default-0/platcheck_12.o -pthread -Wl,--export-dynamic -lffi -lrt -o /tmp/usession-default-0/platcheck_12
[platform:execute] gcc -c -O3 -pthread -fomit-frame-pointer -Wall -Wno-unused -I/home/jmglov/Documents/code/pixie/externals/pypy/rpython/translator/c /tmp/usession-default-0/platcheck_13.c -o /tmp/usession-default-0/platcheck_13.o
[platform:execute] gcc -c -O3 -pthread -fomit-frame-pointer -Wall -Wno-unused -I/home/jmglov/Documents/code/pixie/externals/pypy/rpython/translator/c /home/jmglov/Documents/code/pixie/externals/pypy/rpython/translator/c/src/thread.c -o /tmp/usession-default-0/rpython/translator/c/src/thread.o
[platform:execute] gcc /tmp/usession-default-0/platcheck_13.o /tmp/usession-default-0/rpython/translator/c/src/thread.o -pthread -Wl,--export-dynamic -lrt -o /tmp/usession-default-0/platcheck_13
[platform:execute] gcc -c -O3 -pthread -fomit-frame-pointer -Wall -Wno-unused -I/home/jmglov/Documents/code/pixie/externals/pypy/rpython/translator/c /tmp/usession-default-0/platcheck_14.c -o /tmp/usession-default-0/platcheck_14.o
[platform:execute] gcc -c -O3 -pthread -fomit-frame-pointer -Wall -Wno-unused -I/home/jmglov/Documents/code/pixie/externals/pypy/rpython/translator/c /home/jmglov/Documents/code/pixie/externals/pypy/rpython/translator/c/src/stacklet/stacklet.c -o /tmp/usession-default-0/rpython/translator/c/src/stacklet/stacklet.o
[platform:execute] gcc /tmp/usession-default-0/platcheck_14.o /tmp/usession-default-0/rpython/translator/c/src/stacklet/stacklet.o -pthread -Wl,--export-dynamic -lrt -o /tmp/usession-default-0/platcheck_14
Traceback (most recent call last):
  File "externals/pypy/rpython/bin/rpython", line 20, in <module>
    main()
  File "/home/jmglov/Documents/code/pixie/externals/pypy/rpython/translator/goal/translate.py", line 217, in main
    targetspec_dic, translateconfig, config, args = parse_options_and_load_target()
  File "/home/jmglov/Documents/code/pixie/externals/pypy/rpython/translator/goal/translate.py", line 156, in parse_options_and_load_target
    targetspec_dic = load_target(targetspec)
  File "/home/jmglov/Documents/code/pixie/externals/pypy/rpython/translator/goal/translate.py", line 98, in load_target
    mod = __import__(specname)
  File "target.py", line 311, in <module>
    rt.init()
  File "pixie/vm/rt.py", line 76, in init
    import pixie.vm.c_api
  File "pixie/vm/c_api.py", line 8, in <module>
    @entrypoint('main', [rffi.CCHARP], c_name='pixie_init')
  File "/home/jmglov/Documents/code/pixie/externals/pypy/rpython/rlib/entrypoint.py", line 106, in entrypoint
    raise Exception("entrypoint.entrypoint() is removed because of a bug.  "
Exception: entrypoint.entrypoint() is removed because of a bug.  Remove the 'aroundstate' code in your functions and then call entrypoint_highlevel(), which does that for you.  Another difference is that entrypoint_highlevel() returns the normal Python function, which can be safely called from more Python code.
Makefile:22: recipe for target 'build_with_jit' failed
make: *** [build_with_jit] Error 1
```

**Comment by jmglov** [comment]
The build works fine with pypy-4.0.1. Next I'll try following the advice in the error message and seeing if I can build with the pypy nightly.

**Comment by heyLu** [comment]
Ok, thanks!  You might stumble upon #462 after fixing it, though.

**Comment by jmglov** [comment]
@heyLu I copied your fix for #462 into my PR, and things seem to be working fine with the resulting build.
